Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -35,4 +48,36 @@ void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
     // This is where you can update your variables
     // Hint: There are many different ways that the totalAmpHoursUsed can be updated
     // i.e: Taking a running average of your data values, using most recent data points, etc.
+
+    // The section below is updates the total Amphours used
+    if (dataPoints_ > 0)
+        timeDiff_ = (double)(batteryData_.time.msecsTo(batteryData.time)*0.000000278);
+
+    batteryData_.time = batteryData.time;
+
+    if (dataPoints_ == 0)
+        totalAmpHoursUsed_ = ampHours_;
+    if (dataPoints_ > 0)
+        totalAmpHoursUsed_ += (((batteryData.current + batteryData_.current) /2) * timeDiff_);
```

Comment:
Space between operators.
` totalAmpHoursUsed_ += (((batteryData.current + batteryData_.current) / 2) * timeDiff_);`

Reply:
Also, this one-liner works, but for simpler readability, I would suggest that we split up the calculation.

Reply:
Lastly, this `if` statement should be an `else if`, and can be combined with the other one before it.